### PR TITLE
feat: [Custom widget] Make builder editor section resizeable

### LINKED ISSUE #30821: [enhancement]: [Custom widget] Make the editor section of the custom widget builder resizeable

### PR DESCRIPTION
**Pull request by sbalaji1192**
#### PR fixes following issue(s)
Fixes #30821

#### Type of change

- New feature (non-breaking change which adds functionality)
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced a more dynamic layout for the Custom Widget Builder, including adjustable heights and collapsible rows.
	- Enhanced the Custom Widget Builder's editor and preview panes with resizable capabilities and improved state management.
	- Added persistent state features for the Custom Widget Builder, utilizing local storage for user preferences like row collapse states.
- **Refactor**
	- Streamlined the Custom Widget Builder's codebase for better performance and maintainability.
	- Updated the Custom Widget Builder's styling and layout for a more intuitive user interface.
- **Style**
	- Updated styles for the Custom Widget Builder, including adjustments to headers, borders, and overflow handling for a cleaner look and feel.

### CODE REVIEW

File: app/client/src/pages/Editor/CustomWidgetBuilder/useCustomBuilder.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
```
Comment (somangshu) [inline app/client/src/pages/Editor/CustomWidgetBuilder/useCustomBuilder.tsx:1]: For my context, what is this for?

**Review by somangshu — APPROVED** [review]
Code LGTM. 
Going to test the DP

**Comment by somangshu** [comment]
@sbalaji1192 why is this a chore and has the label skip-changelog. 
From what I understand, We are shipping a feature

File: app/client/src/pages/Editor/CustomWidgetBuilder/useCustomBuilder.tsx (hunk 1 above)
Comment (sbalaji1192) [reply app/client/src/pages/Editor/CustomWidgetBuilder/useCustomBuilder.tsx:1]: I have extracted out some code into a new file for better readability

**Comment by sbalaji1192** [comment]
@somangshu I don't think we should consider this as a feature. its a small enhancement to an existing UI. No new functionality has been added from user perspective.

**Pull request merged**